lib/propshaft/helper.rb

```diff
@@ -4,22 +4,40 @@ def compute_asset_path(path, options = {})
       Rails.application.assets.resolver.resolve(path) || raise(MissingAssetError.new(path))
     end
 
-    # Add an option to call `stylesheet_link_tag` with `:all` to include every css file found on the load path.
+    # Add an option to call `stylesheet_link_tag` with
+    # `:all` to include every css file found on the load path or
+    # `:app` to include every css file found on `app/assets/stylesheets`.
     def stylesheet_link_tag(*sources, **options)
-      if sources.first == :all
+      case sources.first
+      when :all
         super(*all_stylesheets_paths, **options)
+      when :app
+        super(*app_stylesheets_paths, **options)
       else
         super
       end
     end
 
-    # Returns a sorted and unique array of logical paths for all stylesheets in the load path.
-    def all_stylesheets_paths
-      Rails.application.assets.load_path
-        .assets(content_types: [ Mime::EXTENSION_LOOKUP["css"] ])
-        .collect { |css| css.logical_path.to_s }
-        .sort
-        .uniq
-    end
+    private
+      def all_stylesheets_paths
+        stylesheets_paths_for(Rails.application.assets.load_path)
+      end
+
+      def app_stylesheets_paths
+        stylesheets_paths_for(
+          Rails.application.assets.load_path.dup.tap do |load_path|
+            load_path.paths = [ File.join(Rails.root, "app", "assets", "stylesheets") ]
```

Comment:
You should be able to just use `Rails.root.join("app/assets/stylesheets")` here.

Reply:
Done

Reply:
Actually, on second thought, we don't need the load_path or the helper method to do this. We can just do:

```ruby
def app_stylesheet_paths
  Rails.root.join("app/assets/stylesheets").then do |dir|
    dir.glob("**/*.css").collect { |f| f.relative_path_from(dir).to_s }.sort
  end
end
```

Although trying to think you might/probably want to cache this in production.

Reply:
Yeah, the cache question is an issue. Same for :all. We shouldn't be doing file stats on every request. Think we need a way to get this into a proper object with caching that's reset by a sweeper, like LoadPath.

Reply:
Actually maybe caching doesn't matter. Just did a simple test on HEY:

```ruby
haystack(dev)* Benchmark.measure { 1000.times { Rails.root.join("app/assets/stylesheets").then do |dir|
haystack(dev)*     dir.glob("**/*.css").collect { |f| f.relative_path_from(dir).to_s }.sort
haystack(dev)*     end
haystack(dev)> } }.real / 1000
=> 0.00025197141902754083
```

So that's an overhead of 0.2ms. Which probably means that the file system or whatever is already providing the cache. 

Guess it could be worth looking at a more pathological case, maybe? Slow CPU + HDD might well give a different result.

Reply:
That benchmark wasn't correct. This is right:

```ruby
Benchmark.measure { 1000.times { Rails.root.join("app/assets/stylesheets").then { |dir| dir.glob("**/*.scss").collect { |f| f.relative_path_from(dir).to_s } } } }.real / 1000
```

Which sets the price at 1ms for 263 files on a very fast CPU + SSD. That's not great. So back to looking at caching that's stable in production!

Reply:
Which actually gives me wonder whether we should have a generalized cache like this. Stable in production, pass-through in dev/test.